### PR TITLE
Add some basic enanglement tests

### PR DESCRIPTION
**Pull request by oxarbitrage**
I was working with some basic entanglement so i created a few tests. Feel free to discord them if they are too basic or stuff already tested somewhere else.

There are also a few `TODOS`s i added to the code that i can try to fix in the future.

### CODE REVIEW

File: tests/entanglement.rs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        let binary_m = format!("{:02b}", m);
+        assert_eq!(
+            binary_m.chars().nth(0).unwrap().to_digit(2).unwrap(),
+            input.second_bit as u32
+        );
+        assert_eq!(
+            binary_m.chars().nth(1).unwrap().to_digit(2).unwrap(),
+            input.first_bit as u32
+        );
+        // TODO: Positions are reversed, check if there is some sort of endianess happening.
```
Comment (oxarbitrage) [inline tests/entanglement.rs:144]: Maybe related https://github.com/Renmusxd/RustQIP/issues/25 ?

**Comment by Renmusxd** [comment]
Sorry - taking me a while to get to this. I'll consider whether an entanglement unit test could get a little fancier. Right now it seems to just be checking that the hadamard gate works as expected which is already covered by stuff like the CSWAP test. But something more interesting along these lines could be useful to have.

File: tests/entanglement.rs (hunk 1 above)
Comment (Renmusxd) [reply tests/entanglement.rs:144]: You're right that the endianness issue is related to #25 - and the whole thing comes from the original iterators being defined to match how numpy handles outer (/tensor) products. Rewriting the internals may be an undertaking so I'm considering whether it's worthwhile.

**Comment by oxarbitrage** [comment]
No worries, i am just trying to understand how the software works and learning some quantum computing at the same time so probably some code i  send will not be good enough to be included. I think we can close it, i can probably try to make something a bit different. Feel free to send suggestions on what you will like to see in a better entanglement test :)

**Comment by Renmusxd** [comment]
Yeah - I may come up with some good entanglement ideas after the summer. I will be taking a quantum computing course so will likely just implement some of the ideas we talk about as unit tests.

I'll close this for now but reference issues #2 and #7 so this isn't forgotten.

**Pull request closed without merging**